clients/roscpp/src/libros/transport_subscriber_link.cpp

```diff
@@ -52,6 +52,7 @@ TransportSubscriberLink::TransportSubscriberLink()
 TransportSubscriberLink::~TransportSubscriberLink()
 {
   drop();
+  connection_->removeDropListener(dropped_conn_);
```

Comment:
Isn't the `drop()` method already handling this? https://github.com/ros/ros_comm/blob/35c0a105c6d7ed6601f54487462b08743e84b8c2/clients/roscpp/src/libros/transport_subscriber_link.cpp#L237-L244

Reply:
TransportSubscriberLink::~TransportSubscriberLink() -> drop() -> to call connection_->drop(Connection::Destructing);  
This path will not call "connection_->removeDropListener"